File: samples/src/main/java/tk/zielony/carbonsamples/GuidelinesActivity.java
package tk.zielony.carbonsamples;

import android.os.Bundle;

import java.io.Serializable;

import carbon.component.PaddingItem;
import tk.zielony.carbonsamples.guidelines.BasilActivity;
import tk.zielony.carbonsamples.guidelines.ButtonsUsageActivity;
import tk.zielony.carbonsamples.guidelines.CraneActivity;
import tk.zielony.carbonsamples.guidelines.MenusBehaviorActivity;
import tk.zielony.carbonsamples.guidelines.ShrineActivity;

@ActivityAnnotation(title = R.string.guidelinesActivity_title, layout = R.layout.activity_samplelist)
public class GuidelinesActivity extends SampleListActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        initToolbar();

        setItems(new Serializable[]{
                new PaddingItem(getResources().getDimensionPixelSize(R.dimen.carbon_paddingHalf)),
                "Sample screens taken from Material Design guidelines",
                new SampleActivityItem(ButtonsUsageActivity.class),
                new SampleActivityItem(MenusBehaviorActivity.class),
                new SampleActivityItem(ShrineActivity.class, 0, true),
                new SampleActivityItem(CraneActivity.class),
                new SampleActivityItem(BasilActivity.class, 0, true),
                new PaddingItem(getResources().getDimensionPixelSize(R.dimen.carbon_paddingHalf))
        });
    }

}

File: samples/src/main/java/tk/zielony/carbonsamples/GuidelinesActivity.kt
package tk.zielony.carbonsamples

import android.os.Bundle
import tk.zielony.carbonsamples.guidelines.*

@SampleAnnotation(
        titleId = R.string.guidelinesActivity_title,
        layoutId = R.layout.activity_samplelist,
        iconId = R.drawable.ic_android_black_24dp
)
class GuidelinesActivity : SampleListActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        initToolbar()

        setItems(listOf(
                "Sample screens taken from Material Design guidelines",
                SampleActivityItem(ButtonsUsageActivity::class.java),
                SampleActivityItem(MenusBehaviorActivity::class.java),
                SampleActivityItem(ShrineActivity::class.java, true),
                SampleActivityItem(CraneActivity::class.java),
                SampleActivityItem(BasilActivity::class.java, true)
        ))
    }

}

